src/Arcus.BackgroundJobs.CloudEvents/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -119,16 +112,55 @@ public static class IServiceCollectionExtensions
             Guard.NotNullOrWhitespace(subscriptionNamePrefix, nameof(subscriptionNamePrefix), "Requires a non-blank subscription name of the Azure Service Bus Topic subscription, to receive Key Vault events");
             Guard.NotNullOrWhitespace(serviceBusTopicConnectionStringSecretKey, nameof(serviceBusTopicConnectionStringSecretKey), "Requires a non-blank secret key that points to a Azure Service Bus Topic-scoped connection string");
 
-            services.AddServiceBusMessageRouting(serviceProvider =>
+            services.AddServiceBusMessageRouting(ServiceBusImplementationFactory(configureBackgroundJob));
+
+            return services.AddServiceBusTopicMessagePumpWithPrefix(subscriptionNamePrefix, serviceBusTopicConnectionStringSecretKey, configureBackgroundJob);
+        }
+
+        /// <summary>
+        /// Adds a background job to the <see cref="IServiceCollection"/> to receive <see cref="CloudEvent"/>'s.
+        /// </summary>
+        /// <param name="services">The services collection to add the job to.</param>
+        /// <param name="serviceBusNamespace">The Service Bus namespace to connect to. This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="topicName">The name of the Azure Service Bus Topic to process.</param>
+        /// <param name="subscriptionNamePrefix">The name of the Azure Service Bus subscription that will be created to receive <see cref="CloudEvent"/>'s.</param>
+        /// <param name="clientId">
+        ///     The client ID to authenticate for a user assigned managed identity. More information on user assigned managed identities cam be found here:
+        ///     <see href="https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#how-a-user-assigned-managed-identity-works-with-an-azure-vm" />.
+        /// </param>
+        /// <param name="configureBackgroundJob">The capability to configure additional options on how the CloudEvents background job should behave.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="services"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="serviceBusNamespace"/>, the <paramref name="topicName"/>, or the <paramref name="subscriptionNamePrefix"/> is blank.
+        /// </exception>
+        public static ServiceBusMessageHandlerCollection AddCloudEventBackgroundJobUsingManagedIdentity(
+            this IServiceCollection services,
+            string serviceBusNamespace,
+            string topicName,
+            string subscriptionNamePrefix,
+            string clientId = null,
+            Action<IAzureServiceBusTopicMessagePumpOptions> configureBackgroundJob = null)
```

Comment:
Hmm, this is a different order as the one were we register the message pump. For better signature detection, we might use the same order (`topicName`, `subscriptionNamePrefix`, `serviceBusNamespace`, ...).

Reply:
Changed the order of the parameters and updated the docs as well.